ecs-agent/api/container/restart/restart_tracker.go

```diff
@@ -0,0 +1,91 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package restart
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	api "github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+	apicontainerstatus "github.com/aws/amazon-ecs-agent/ecs-agent/api/container/status"
+
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+type RestartTracker struct {
+	RestartCount  int       `json:"restartCount,omitempty"`
+	LastStartedAt time.Time `json:"lastStartedAt,omitempty"`
+	restartPolicy *api.ContainerRestartPolicy
+	lock          sync.RWMutex
+}
+
+func NewRestartTracker(restartPolicy *api.ContainerRestartPolicy) *RestartTracker {
+	return &RestartTracker{
+		restartPolicy: restartPolicy,
+	}
+}
+
+func (rt *RestartTracker) GetRestartEnabled() bool {
```

Comment:
Why is this method necessary? Isn't this information only needed by the restart tracker since it is the one making the decision if a restart should be performed or not?

Reply:
The method is not strictly necessary but may be helpful for implementers to check. e.g. if restart policy is configured but not enabled we may choose not to restart.

Reply:
>  e.g. if restart policy is configured but not enabled we may choose not to restart.

But the restart tracker should be the one making that choice right? I don't think we should expose this publicly unless there is a need to (i.e., keep things private unless there is a reason to make them public).

Reply:
Sure, if need be we can follow this up and re-add. I have removed now.